internal/test/e2e/run.go

```diff
@@ -203,19 +205,26 @@ func splitTests(testsList []string, conf ParallelRunConf) []instanceRunConf {
 
 	vsphereTestsRe := regexp.MustCompile(vsphereRegex)
 	privateNetworkTestsRe := regexp.MustCompile(`^.*(Proxy|RegistryMirror).*$`)
+	multiClusterTestsRe := regexp.MustCompile(`^.*Multicluster.*$`)
 
 	runConfs := make([]instanceRunConf, 0, conf.MaxInstances)
 	ipman := newE2EIPManager(os.Getenv(cidrVar), os.Getenv(privateNetworkCidrVar))
 
 	testsInCurrentInstance := make([]string, 0, testPerInstance)
 	for _, testName := range testsList {
 		testsInCurrentInstance = append(testsInCurrentInstance, testName)
-		ip := ""
+		var ips framework.E2EIPPool
 		if privateNetworkTestsRe.MatchString(testName) {
-			ip = ipman.getPrivateIP()
+			ips = ipman.reservePrivateIPPool(1)
 		} else if vsphereTestsRe.MatchString(testName) {
-			ip = ipman.getIP()
+			if multiClusterTestsRe.MatchString(testName) {
```

Comment:
Hard coding to 2 ips for now in order to test this initially since we only have 1 multicluster test. Ideally, we need to find a way for each test to notify how many ips it requires. 

We either need to pass number in test name and parse it as we do now via regex or create some type of mapping. I'm open to ideas and suggestions.